### ISSUE TITLE
Obfuscated modules import hangs on Apple silicon using pyarmor 3.6. 

### ISSUE THREAD

**Issue by fdv1** [dashingsoft/pyarmor#693]
I use the following commands:

pyarmor runtime -O dist --enable-suffix --platform darwin.aarch64
pyarmor obfuscate --with-runtime @dist reflect/__init__.py darwin.aarch64

Could you please assist? thanks!

**Comment by jondy** [comment]
There are several familar issues here, search it and find the solution.

**Comment by fdv1** [comment]
Hey Jondy, not sure what you mean by "familiar issues". Do you mean that this has already been reported? or shall look on your website to see how to troubleshoot this kind of issues? (although it looks almost certainly a pyarmor issue since other platforms work great!)

**Comment by jondy** [comment]
Francois du Vignaud ***@***.***> writes:

> Hey Jondy, not sure what you mean by "familiar issues". Do you mean that this
> has already been reported?

Yes, this issue has been reported by many times in the issues.

**Comment by fdv1** [comment]
Sorry for being dumb but I can't see any bug describing the behaviour I'm experiencing, ie: crash at import on apple silicon only. Would you mind letting me(us) know which issue you have in mind? As I mentioned already it does work well on other platforms...

**Comment by jondy** [comment]
One possible reason is the pyarmor runtime binary file not code signed, so try to use command `codesign` the runtime file `_pytransform.dylib`. In the latest version, it has been also code signed, but the old version not.

**Comment by fdv1** [comment]
Sorry I thought you were suggesting to look into the git issues to find a solution to this problem. Anyway this prompted me to revamp my modules so only one of them is encrypted now, so this is no longer blocking me. I must confess I am not completely familiar with the whole pyarmor interface but I think it would be nice to display warning when you detect a potential misuse if that's possible.

**Comment by fdv1** [comment]
Hi Jondy,
As I wrote earlier I noticed that I could generate universal binaries by not sharing the runtime. I realize now the issue might not be related to that but rather to the binary being shipped with the encrypted dist. The following produces working packages:
INFO     Copying /Users/francois/.pyarmor/platforms/darwin/aarch64/0/_pytransform.dylib
whereas the following one hangs at import:
INFO     Copying /Users/francois/.pyarmor/platforms/darwin/aarch64/3/_pytransform.dylib
The reason why it would pick one or the other aren't entirely clear to me. Basically I end up picking the buggy version when spawning a subprocess from a different python version. (my build script is in python and spawn ad hoc python subproceses to encrypt each version). 
Hope this help,
Francois
